### ISSUE TITLE
Consider using c.ClientIP() from Gin in middleware

### ISSUE THREAD

**Issue by kdar** [ulule/limiter#26]
The issue with using utils.GetIPKey() when using the Gin middleware is that it does not respect the engine.ForwardedByClientIP option (https://godoc.org/github.com/gin-gonic/gin#Engine).

This is a security issue if you're not running your server behind a proxy.

**Comment by novln** [comment]
Hello,

I'm sorry but I'm not sure to follow you. If `engine.ForwardedByClientIP` is `true` we should ignore `X-Forwarded-For` and `X-Real-IP` ? Or the other way around ?

Anyway, thank you for your feedback: highly appreciated.

Cheers,

**Comment by kdar** [comment]
If `engine.ForwardByClientIP` is set to `true`, then `X-Forwarded-For` and `X-Real-IP` are used. If it's set to `false`, they are ignored.

Relevant code in Gin:

```
func (c *Context) ClientIP() string {
	if c.engine.ForwardedByClientIP {
		clientIP := c.requestHeader("X-Forwarded-For")
		if index := strings.IndexByte(clientIP, ','); index >= 0 {
			clientIP = clientIP[0:index]
		}
		clientIP = strings.TrimSpace(clientIP)
		if len(clientIP) > 0 {
			return clientIP
		}
		clientIP = strings.TrimSpace(c.requestHeader("X-Real-Ip"))
		if len(clientIP) > 0 {
			return clientIP
		}
	}

	if c.engine.AppEngine {
		if addr := c.requestHeader("X-Appengine-Remote-Addr"); addr != "" {
			return addr
		}
	}

	if ip, _, err := net.SplitHostPort(strings.TrimSpace(c.Request.RemoteAddr)); err == nil {
		return ip
	}

	return ""
}
```

**Comment by novln** [comment]
Fixed by #27

Thank you for your feedback and input.
